src/Microsoft.ML.Transforms/FourierDistributionSampler.cs

```diff
@@ -5,205 +5,270 @@
 using System;
 using Microsoft.ML;
 using Microsoft.ML.CommandLine;
-using Microsoft.ML.EntryPoints;
+using Microsoft.ML.Data;
 using Microsoft.ML.Internal.Utilities;
 using Microsoft.ML.Model;
-using Microsoft.ML.Transforms;
+using Microsoft.ML.Numeric;
+using Microsoft.ML.Transforms.Projections;
 
-[assembly: LoadableClass(typeof(GaussianFourierSampler), typeof(GaussianFourierSampler.Options), typeof(SignatureFourierDistributionSampler),
-    "Gaussian Kernel", GaussianFourierSampler.LoadName, "Gaussian")]
+[assembly: LoadableClass(typeof(GaussianKernel), typeof(GaussianKernel.Options), typeof(SignatureRngGenerator),
+    "Gaussian Kernel", GaussianKernel.LoadName, "Gaussian")]
 
-[assembly: LoadableClass(typeof(LaplacianFourierSampler), typeof(LaplacianFourierSampler.Options), typeof(SignatureFourierDistributionSampler),
-    "Laplacian Kernel", LaplacianFourierSampler.RegistrationName, "Laplacian")]
+[assembly: LoadableClass(typeof(LaplacianKernel), typeof(LaplacianKernel.Options), typeof(SignatureRngGenerator),
+    "Laplacian Kernel", LaplacianKernel.LoadName, "Laplacian")]
 
 // This is for deserialization from a binary model file.
-[assembly: LoadableClass(typeof(GaussianFourierSampler), null, typeof(SignatureLoadModel),
-    "Gaussian Fourier Sampler Executor", "GaussianSamplerExecutor", GaussianFourierSampler.LoaderSignature)]
+[assembly: LoadableClass(typeof(GaussianKernel.RandomNumberGenerator), null, typeof(SignatureLoadModel),
+    "Gaussian Fourier Sampler Executor", "GaussianSamplerExecutor", GaussianKernel.RandomNumberGenerator.LoaderSignature)]
 
 // This is for deserialization from a binary model file.
-[assembly: LoadableClass(typeof(LaplacianFourierSampler), null, typeof(SignatureLoadModel),
-    "Laplacian Fourier Sampler Executor", "LaplacianSamplerExecutor", LaplacianFourierSampler.LoaderSignature)]
+[assembly: LoadableClass(typeof(LaplacianKernel.RandomNumberGenerator), null, typeof(SignatureLoadModel),
+    "Laplacian Fourier Sampler Executor", "LaplacianSamplerExecutor", LaplacianKernel.RandomNumberGenerator.LoaderSignature)]
 
-// REVIEW: Roll all of this in with the RffTransform.
-namespace Microsoft.ML.Transforms
+namespace Microsoft.ML.Transforms.Projections
 {
     /// <summary>
-    /// Signature for an IFourierDistributionSampler constructor.
+    /// Signature for a <see cref="KernelBase"/> constructor.
     /// </summary>
     [BestFriend]
-    internal delegate void SignatureFourierDistributionSampler(float avgDist);
+    internal delegate void SignatureRngGenerator();
 
-    public interface IFourierDistributionSampler : ICanSaveModel
+    /// <summary>
+    /// This class indicates which kernel should be approximated by the <see cref="RandomFourierFeaturizingTransformer"/>.
+    /// <seealso cref="RandomFourierFeaturizingEstimator"/>.
+    /// </summary>
+    public abstract class KernelBase
```

Comment:
We should add a `private protected` constructor to this class, to explicitly tell external devs that they can't inherit from it.
The way it is now, it has an implicit `protected` constructor, which makes it seem like someone could inherit from it. But if they try, they will get a compile error saying it doesn't implement the abstract methods below. Having the `private protected` constructor shows that we intend that no one outside of ML.NET should inherit from this class.